src/run_build.py

```diff
@@ -69,12 +70,20 @@ def main() -> int:
                 builder.export_artifacts(build_recorder)
             except:
                 logging.error(f"Error building {component.name}, retry with: {args.component_command(component.name)}")
-                raise
+                if component.name in ['OpenSearch', 'common-utils', 'OpenSearch-Dashboards']:
```

Comment:
I am not sure if I like this approach.
There are so many dependencies between plugins, such as AD based on JS/ML, neural based on KNN, and more.

I would say just add everything to the failed list, and the users can figure out the issue by checking failed list, instead of hard coding which one should hard fail which one should be added to list.

Reply:
I would also say let us add a param to `hard fail` no matter what, so user can choose where 1 component fail immediately fail the entire thing, or get added to the fail list. Which is similar to your approach but does not have a hardcode list, but depending on user input params.